Parsers/ASimNetworkSession/Parsers/vimNetworkSessionCheckPointFirewall.yaml

```diff
@@ -0,0 +1,329 @@
+          , "110","Compaq-Peer"
+          , "111","IPX-in-IP"
+          , "112","VRRP"
+          , "113","PGM"
+          , "114",""
+          , "115","L2TP"
+          , "116","DDX"
+          , "117","IATP"
+          , "118","STP"
+          , "119","SRP"
+          , "120","UTI"
+          , "121","SMP"
+          , "122","SM (deprecated)"
+          , "123","PTP"
+          , "124","ISIS over IPv4"
+          , "125","FIRE"
+          , "126","CRTP"
+          , "127","CRUDP"
+          , "128","SSCOPMCE"
+          , "129","IPLT"
+          , "130","SPS"
+          , "131","PIPE"
+          , "132","SCTP"
+          , "133","FC"
+          , "134","RSVP-E2E-IGNORE"
+          , "135","Mobility Header"
+          , "136","UDPLite"
+          , "137","MPLS-in-IP"
+          , "138","manet"
+          , "139","HIP"
+          , "140","Shim6"
+          , "141","WESP"
+          , "142","ROHC"
+          , "143","Ethernet"
+          , "253",""
+          , "254",""
+          , "255","Reserved"];
+      let DirectionLookup=datatable(conn_direction:string,NetworkDirection:string)
+      [
+          "Incoming","Inbound", 
+          "Outgoing","Outbound", 
+          "Internal","Local"];
+      let ActionLookup=datatable(DvcOriginalAction:string,DvcAction:string,EventResult:string,EventSeverity:string)
+      [
+          "Accept","Allow","Success","Informational",
+          "Allow","Allow","Success","Informational",
+          "Drop","Drop","Failure","Low",
+          "Reject","Deny","Failure","Low",
+          "Encrypt","Encrypt","Success","Informational",
+          "Decrypt","Decrypt","Success","Informational",
+          "Bypass","Allow","Success","Informational",
+          "Block","Deny","Failure","Low",
+          "","","NA","Informational"
+       ];
+        let NWParser=(
+            starttime:datetime=datetime(null), 
+            endtime:datetime=datetime(null),
+            srcipaddr_has_any_prefix:dynamic=dynamic([]), 
+            dstipaddr_has_any_prefix:dynamic=dynamic([]), 
+            ipaddr_has_any_prefix:dynamic=dynamic([]),
+            dstportnumber:int=int(null), 
+            hostname_has_any:dynamic=dynamic([]), 
+            dvcaction:dynamic=dynamic([]), 
+            eventresult:string='*', 
+            disabled:bool=false)
+        {
+        let src_or_any = set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+        let dst_or_any = set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix);
+        CommonSecurityLog
+        | where not(disabled)
+        | where
+                        (isnull(starttime) or TimeGenerated >= starttime) 
+                        and (isnull(endtime) or TimeGenerated <= endtime)
+        | where 
+                        array_length(hostname_has_any) == 0
+        | where DeviceVendor=="Check Point" and DeviceProduct=="VPN-1 & FireWall-1"
+        | where (isnull(dstportnumber) or (DestinationPort == dstportnumber))
+        | where ((eventresult == "*") 
```

Comment:
We modified the logic for EventResult to use DeviceAction. What I suggest is to do the action lookup in line 256 and check the condition immediately after, instead of this line and the lines in 302/3

Reply:
Updated with DeviceAction